src/include/add-on/add-on-workflow.rb

```diff
@@ -1027,6 +1027,9 @@ def RunAddOnMainDialog(enable_back, enable_next, enable_abort, back_button, next
           Wizard.SetTitleIcon("yast-addon")
           ret2 = RunWizard()
 
+          break if ret2 == :back
```

Comment:
This `break` here avoid the right propagation of the user input from the sub-workflow (aka `ret2`) to the main workflow (aka `ret`).

Fixed in #89